rusoto/credential/src/lib.rs

```diff
@@ -282,28 +361,121 @@ fn in_ten_minutes() -> DateTime<Utc> {
 }
 
 /// Reduces Boilerplate on getting json values. Wraps `serde_json::Value.get(key)`.
-fn extract_string_value_from_json(json_object: &Value, key: &str) -> Result<String, CredentialsError> {
+fn extract_string_value_from_json(
+    json_object: &Value,
+    key: &str,
+) -> Result<String, CredentialsError> {
     match json_object.get(key) {
-        Some(v) => Ok(v.as_str().expect(&format!("{} value was not a string", key)).to_owned()),
-        None => Err(CredentialsError::new(format!("Couldn't find {} in response.", key))),
+        Some(v) => Ok(
+            v.as_str()
+                .expect(&format!("{} value was not a string", key))
+                .to_owned(),
+        ),
+        None => Err(CredentialsError::new(
+            format!("Couldn't find {} in response.", key),
+        )),
     }
 }
 
 /// Parses the response from an AWS Metadata Service, either from an IAM Role, or a Container.
 fn parse_credentials_from_aws_service(response: &str) -> Result<AwsCredentials, CredentialsError> {
     let json_object: Value = match json_from_str(response) {
         Ok(v) => v,
-        Err(_) => return Err(CredentialsError::new("Couldn't parse credentials response body.")),
+        Err(_) => {
+            return Err(CredentialsError::new(
+                "Couldn't parse credentials response body.",
+            ))
+        }
     };
 
     let access_key_id = try!(extract_string_value_from_json(&json_object, "AccessKeyId"));
-    let secret_access_key = try!(extract_string_value_from_json(&json_object, "SecretAccessKey"));
+    let secret_access_key = try!(extract_string_value_from_json(
+        &json_object,
+        "SecretAccessKey",
+    ));
     let token = try!(extract_string_value_from_json(&json_object, "Token"));
     let expiration = try!(extract_string_value_from_json(&json_object, "Expiration"));
 
     let expiration = try!(expiration.parse());
 
-    Ok(AwsCredentials::new(access_key_id, secret_access_key, Some(token), expiration))
+    Ok(AwsCredentials::new(
+        access_key_id,
+        secret_access_key,
+        Some(token),
+        expiration,
+    ))
+}
+
+/// Makes an Async Request with a timeout. Defaults to 30 seconds.
+fn make_request(
+    mut core: Core,
+    uri: Uri,
+    duration: Option<StdDuration>,
+) -> Result<String, hyper::Error> {
+
+    let handle = core.handle();
+    let connector = HttpsConnector::new(4, &handle);
+    if connector.is_err() {
```

Comment:
Checking for `is_err()` then calling `unwrap()` on the same item later tripped me up when reviewing this.  Is it possible to switch this to `let connector = match...` and use that pattern instead?  I know it can be tricky sometime but it's very clear we don't call `unwrap()` on anything.

Reply:
Yea I can do this. I generally prefer unwrap, but don't mind doing it for the sake of not using unwrap.